### PR TITLE
fix: fix listen error address mistake

### PR DESCRIPTION
**Pull request by driftluo**
There is a wrong output in listen error

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -779,7 +786,7 @@ where
                             &mut self.service_context,
                             ServiceError::ListenError {
                                 error: Error::RepeatedConnection(context.inner.id),
-                                address,
+                                address: listen_addr.expect("listen address must be exist"),
```
Comment (doitian) [inline src/service.rs]:
```suggestion
                                address: listen_addr.expect("listen address must exist"),
```

Comment (doitian) [reply src/service.rs]: Why we can ensure `listen_addr` is not None, I see several places that call `session_open` or `handshake` with None.

Comment (driftluo) [reply src/service.rs]: There are two situations for a handshake:
1. self is listener, session type is inbound
2. self is dialer, session type is outboud

When self is listener, session is inbound, listen address must exist

Comment (doitian) [reply src/service.rs]: Is it possible to change `SessionType::Inbound` to ` SessionType::Inbound { listen_addr: MultiAddr }` ?

Comment (doitian) [reply src/service.rs]: If not, we can add an assert that when session type is inbound, `listen_addr` must be Some.

Comment (driftluo) [reply src/service.rs]: It's a breaking change. 

Users need to know which listen port the session comes from?

Comment (driftluo) [reply src/service.rs]: > If not, we can add an assert that when session type is inbound, `listen_addr` must be Some.

I think `expect` is the same as assert on here

**Pull request merged**